src/platform/nrfconnect/FactoryDataProvider.cpp

```diff
@@ -24,7 +24,15 @@
 #include <platform/Zephyr/ZephyrConfig.h>
 #endif
 
-#include <zephyr/logging/log.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#ifdef CONFIG_CHIP_CRYPTO_PSA
+#include <lib/support/ScopedBuffer.h>
+#include <psa/crypto.h>
+#include <zephyr/drivers/flash.h>
+
+static const struct device * const kFlashDev = DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
```

Comment:
This seems to assume the factory data in external flash is not supported. As more and more factory data features are not supported using external flash I think it arguable if we need the external flash support at all in the current state (just saying, not asking for any changes in this PR :)).

Reply:
Yeah, I completely agree with that, I think we shouldn't suggest using factory data in an external flash at all :)